src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -173,8 +173,9 @@ public final Connection getConnection(final long hardTimeout) throws SQLExceptio
                timeout = hardTimeout - clockSource.elapsedMillis(startTime);
             }
             else {
-               metricsTracker.recordBorrowStats(poolEntry, startTime);
-               return poolEntry.createProxyConnection(leakTask.start(poolEntry), now);
+               poolEntry.lastBorrowed = now;
+               metricsTracker.recordBorrowStats(poolEntry, startTime, now);
+               return poolEntry.createProxyConnection(leakTask.schedule(poolEntry, houseKeepingExecutorService), now);
```

Comment:
it would be cumbersome to add check in PoolEntry#toString() ...need reference to config etc
moreover, I do not see any side effect of initializing lastBorrowed in both cases this way.